tests/Integration/Behaviour/Features/Context/Domain/Product/AbstractProductFeatureContext.php

```diff
@@ -43,6 +43,36 @@
 
 abstract class AbstractProductFeatureContext extends AbstractDomainFeatureContext
 {
+    /**
+     * Transform url from behat test into a proper one, expected value looks like this:
+     *   http://myshop.com/img/p/{image1}-small_default.jpg
+     *
+     * Where image1 is the reference to the image id in the shared storage, it allows to get the image
+     * id and correctly rebuild the url into something like this:
+     *  http://myshop.com/img/p/4/5/45-small_default.jpg
+     *
+     * @param string $imageUrl
+     *
+     * @return string
+     */
+    protected function getRealImageUrl(string $imageUrl): string
+    {
+        // Get image reference which is integrated in image url
+        preg_match('_\{(.+)\}_', $imageUrl, $matches);
```

Comment:
You're also able to use preg_replace_callback :)

Reply:
I could but it's not that straight forward, because I catch what's inside the mustaches `{image1}` (so `image1` here) but I need to replace the mustache plus its inside So I'd need to catch the whole mustache think and perform another preg_match inside my callback to get its content, so in the end, it's not really simpler
This one works fine for now, so I think it's enough 🙂